Etg.Yams.Test/Etg.Yams.Test/Storage/LocalDeploymentRepository.cs

```diff
@@ -0,0 +1,109 @@
+            return Task.FromResult(new DeploymentConfig(data));
+        }
+
+        public Task PublishDeploymentConfig(DeploymentConfig deploymentConfig)
+        {
+            File.WriteAllText(_deploymentConfigPath, deploymentConfig.RawData());
+            return Task.CompletedTask;
+        }
+
+        public Task UploadApplicationBinaries(AppIdentity appIdentity, string localPath, ConflictResolutionMode conflictResolutionMode)
+        {
+            if (FileUtils.DirectoryDoesntExistOrEmpty(localPath))
+            {
+                throw new BinariesNotFoundException(
+                    $"Binaries were not be uploaded because they were not found at the given path {localPath}");
+            }
+
+            string destPath = GetBinariesPath(appIdentity);
+            bool binariesExist = FileUtils.DirectoryDoesntExistOrEmpty(destPath);
+            if (binariesExist)
+            {
+                if (conflictResolutionMode == ConflictResolutionMode.DoNothingIfBinariesExist)
+                {
+                    return Task.CompletedTask;
+                }
+                if (conflictResolutionMode == ConflictResolutionMode.FailIfBinariesExist)
+                {
+                    throw new DuplicateBinariesException();
+                }
+            }
+            return FileUtils.CopyDir(_path, localPath, true);
+        }
+
+        private string GetBinariesPath(AppIdentity appIdentity)
+        {
+            return Path.Combine(_path, GetDeploymentRelativePath(appIdentity));
+        }
+
+        private static string GetDeploymentRelativePath(AppIdentity appIdentity)
+        {
+            return Path.Combine(appIdentity.Id, appIdentity.Version.ToString());
+        }
+
+        public Task DeleteApplicationBinaries(AppIdentity appIdentity)
+        {
+            string path = GetBinariesPath(appIdentity);
+            if (FileUtils.DirectoryDoesntExistOrEmpty(path))
+            {
+                throw new BinariesNotFoundException(
+                    $"Cannot delete binaries for application {appIdentity} because they were not found");
+            }
+            Directory.Delete(path, true);
+            return Task.CompletedTask;
+        }
+
+        public Task<bool> HasApplicationBinaries(AppIdentity appIdentity)
+        {
+            string path = GetBinariesPath(appIdentity);
+            return Task.FromResult(!FileUtils.DirectoryDoesntExistOrEmpty(path));
+        }
+
+        public async Task DownloadApplicationBinaries(AppIdentity appIdentity, string localPath, ConflictResolutionMode conflictResolutionMode)
+        {
+            bool exists = !FileUtils.DirectoryDoesntExistOrEmpty(localPath);
+            if (exists)
+            {
+                if (conflictResolutionMode == ConflictResolutionMode.DoNothingIfBinariesExist)
+                {
+                    return;
+                }
+                if (conflictResolutionMode == ConflictResolutionMode.FailIfBinariesExist)
```

Comment:
Sure, the old version is usually kept in Storage (but can be cleanup if someone choose to do so). This Api doesn't have that knowledge, it's just an Api that allows you to publish binaries and choose what happens if the binaries are already there.